packages/@aws-cdk/aws-events-targets/lib/log-group-resource-policy.ts

```diff
@@ -0,0 +1,50 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as cdk from '@aws-cdk/core';
+import * as cr from '@aws-cdk/custom-resources';
+
+/**
+ * Construction properties for LogGroupResourcePolicy
+ */
+export interface LogGroupResourcePolicyProps {
+  /**
+   * The log group resource policy name
+   */
+  readonly policyName: string;
```

Comment:
does this need to be required? - I think there are a few places in the repo where we automatically generate if not provided (by using the logical ID of the policy resource)

Reply:
I can't get the logical ID until the parent constructor is called (`super()`). And I have to pass the `policyName` as part of the parent constructor parameters. Tried `scope.node.uniqueId()` but it didn't work. I'm afraid it needs to be mandatory.

Reply:
gotcha, it's worth a comment so this context isn't lost

Reply:
So, I didn't know about `cdk.Lazy.stringValue`. Using it, I can generate a uniqueId as a default `policyName`.

```ts
let policyName = props.policyName || cdk.Lazy.stringValue({ produce: () => cdk.Names.uniqueId(this) });
```

I updated the code accordingly and removed the value I had initially created for the policyName!